igb/repl.go

```diff
@@ -327,13 +352,98 @@ func handleParserError(e *parserErr.Error, igb *iGb) {
 
 // Polymorphic helper functions --------------------------------------------
 
+// Returns true if indentation continues.
+func (igb *iGb) continueMultiLineQuote(cdq bool) bool {
+	if cdq { // end multi-line double quote
+		igb.qsm.Event(NoMultiLineQuote)
+		if igb.nqsm.Is(NestedQuote) { // end nested-quote
+			igb.nqsm.Event(NoNestedQuote)
+			igb.cmds = append(igb.cmds, igb.lines)
+			println(prompt(igb.indents) + igb.lines)
+			igb.rl.SetPrompt(prompt2 + indent(igb.indents))
+			return true
+		} else { // exit multi-line double quote
+			igb.cmds = append(igb.cmds, igb.lines)
+			igb.rl.SetPrompt(prompt1)
+			igb.sm.Event(waitEnded)
+			return false
+		}
+	} else { // continue multi-line double quote
+		println(prompt2 + igb.lines)
+		igb.rl.SetPrompt(prompt2)
+		igb.cmds = append(igb.cmds, igb.lines)
+		return true
+	}
+}
+
+func (igb *iGb) eraseBuffer() {
+	igb.indents = 0
+	igb.rl.SetPrompt(prompt1)
+	igb.sm.Event(waitExited)
+	igb.qsm.Event(NoMultiLineQuote)
+	igb.cmds = nil
+	igb.caseBlock = false
+	igb.firstWhen = false
+}
+
+// Prints and add an indent.
+func (igb *iGb) printLineAndIndentRight() {
+	println(prompt(igb.indents) + indent(igb.indents) + igb.lines)
+	igb.indents++
+	igb.rl.SetPrompt(prompt(igb.indents) + indent(igb.indents))
+	igb.cmds = append(igb.cmds, igb.lines)
+}
+
+// Starts multiple line quotation.
+func (igb *iGb) startMultiLineQuote() {
+	if igb.sm.Is(Waiting) {
+		igb.nqsm.Event(NestedQuote)
+	} else {
+		igb.sm.Event(Waiting)
+	}
+	igb.cmds = append(igb.cmds, igb.lines)
+	println(prompt(igb.indents) + indent(igb.indents) + igb.lines)
+	igb.rl.SetPrompt(prompt2)
+}
+
 // Other helper functions --------------------------------------------------
 
+// filterInput just ignores Ctrl-z.
+func filterInput(r rune) (rune, bool) {
+	switch r {
```

Comment:
we can just use `if` since there's only one case

Reply:
updated